src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -989,6 +1000,20 @@ std::optional<UInt64> MergeTreeData::totalRowsByPartitionPredicateImpl(
     return res;
 }
 
+void MergeTreeData::checkSuspiciousIndices(const ASTFunction & index_function) const
```

Comment:
This could be a simple static utility function (i.e. no need to make it a member function of `MergeTreeData`).